### DISCUSSION TITLE
How to recognize datetime type when generating model from JSON

### DISCUSSION THREAD

**Discussion by joelhjalmarsson** [koxudaxi/datamodel-code-generator#2174] [Q&A]
I want to generate a pydantic model from a JSON file, not a schema.

Is it possible to create the JSON in such a way that the code generator can recognize that a field should be of type "datetime" in the resulting model?

**Comment by koxudaxi** [comment]
Not currently possible. The genson library used for raw JSON inference doesn't detect datetime formats.
Use JSON Schema with "format": "date-time" instead for precise type control.
